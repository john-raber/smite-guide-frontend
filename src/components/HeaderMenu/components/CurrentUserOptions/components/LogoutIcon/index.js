import React from "react";
import { connect } from "react-redux";
import { NavLink } from "reactstrap";
import { FaSignOutAlt } from "react-icons/fa";

import { logoutUser } from "../../../../../../scenes/Login/services/currentUser/actions/index";

const LogoutIcon = ({ logoutUser }) => {
  return (
    <NavLink href="/login" onClick={logoutUser}>
      Logout <FaSignOutAlt color="white" />
    </NavLink>
  );
};

export default connect(
  null,
  { logoutUser }
)(LogoutIcon);
