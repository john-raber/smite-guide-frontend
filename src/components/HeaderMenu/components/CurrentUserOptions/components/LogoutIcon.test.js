import React from "react";
import { shallow } from "enzyme";

import { LogoutIcon } from "./LogoutIcon/index";

describe("LogoutIcon", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<LogoutIcon />);

    expect(wrapper.find(".logoutIcon").length).toEqual(1);
  });

  it("matches its snapshot", () => {
    const wrapper = shallow(<LogoutIcon />);

    expect(wrapper).toMatchSnapshot();
  });
});
