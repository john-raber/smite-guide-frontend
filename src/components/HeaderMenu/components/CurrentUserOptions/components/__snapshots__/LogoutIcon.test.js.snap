// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutIcon matches its snapshot 1`] = `
<NavLink
  className="logoutIcon"
  href="/login"
  tag="a"
>
  Logout 
  <FaSignOutAlt
    color="white"
  />
</NavLink>
`;
